### PR TITLE
Use nicer code examples in the component guide

### PR DESCRIPTION
**Pull request by kdoh**
Spruces up our code examples in the component guide to use something more similar to our embed code snippets and sql editor experience (read only as you'd expect). Also lets you copy the code snippet which might prove handy for the less complicated examples like buttons or checkboxes.

screen shot 2018-03-26 at 11 27 40 pm

ToDos
- [x] Proper syntax highlighting

### CODE REVIEW

**Comment by tlrobinson** [comment]
Nice.

One thing I've been meaning to do is make the code samples and props descriptions collapsible, and default to collapsed on the main listing page so it's easier to scan. On the sub pages they could be expanded by default.

Thoughts?

**Comment by kdoh** [comment]
I'm actually wondering if they need to be shown in one long list at all, especially as we add more and more of our existing library to this. It's kinda hard to parse and I'm not sure collapsing is worth the code it would take instead of just having you click on the item you want to view the proper thing in full.

**Comment by tlrobinson** [comment]
I think it's kind of nice to have an overview of the available components, but perhaps it should be limited to a single example of each on the main page. I'm also OK with getting rid of it especially now that there's a sidebar to make browsing easy.

**Comment by tlrobinson** [comment]
BTW I think it would be nicer if the sidebar linked to `/_internal/components/:componentName` instead of `/_internal/components#:componentName`

**Comment by kdoh** [comment]
Yeah agreed. I think I'll leave the route / list cleanup to a follow up branch. Do you have any tips on how to get syntax highlighting working in Ace? I'm having a bit of trouble understanding how to get it to highlight the JSX snippets here to make things easier to parse.

File: frontend/src/metabase/internal/components/ComponentsApp.jsx [hunk 1]
```diff
@@ -101,9 +103,18 @@ export default class ComponentsApp extends Component {
                             <div className="p2 bordered flex align-center flex-full">
                               <div className="full">{element}</div>
                             </div>
-                            <div className="border-left border-right border-bottom text-code">
-                              <div className="p1">
-                                {reactElementToJSXString(element)}
+                            <div className="relative">
+                              <AceEditor
+                                value={reactElementToJSXString(element)}
+                                mode={"jsx"}
```
Comment (tlrobinson) [inline frontend/src/metabase/internal/components/ComponentsApp.jsx]: I think this needs to be `ace/mode/jsx` according to https://github.com/metabase/metabase/blob/master/frontend/src/metabase/public/lib/code.js#L33
